### PR TITLE
ref(issue-search): clean up search code to allow for more customizeable behavior depending on GroupCategory type

### PR DESCRIPTION
**Pull request by barkbarkimashark**
The current issue search code weaved common shared parameters targeting different data sources and data source specific parameters together to build multiple search queries. This resulted in hard to read code that forked and joined parameters together.

This PR refactors the search code to branch off according to the GroupCategory and follow a single codepath for easier readability. I also introduced an extra step for mutating the search_filters based on GroupCategory.

### CODE REVIEW

File: src/sentry/search/snuba/backend.py [hunk 1]
```diff
@@ -533,6 +534,28 @@ def _get_queryset_conditions(
...
+                                    GroupType.PERFORMANCE_N_PLUS_ONE.value,
+                                    GroupType.PERFORMANCE_SLOW_SPAN.value,
+                                    GroupType.PERFORMANCE_SEQUENTIAL_SLOW_SPANS.value,
+                                    GroupType.PERFORMANCE_LONG_TASK_SPANS.value,
+                                    GroupType.PERFORMANCE_RENDER_BLOCKING_ASSET_SPAN.value,
+                                    GroupType.PERFORMANCE_DUPLICATE_SPANS.value,
+                                    GroupType.PERFORMANCE_N_PLUS_ONE_DB_QUERIES.value,
+                                )
+                            )
+                            & Q(message__icontains=query)
```
Comment (barkbarkimashark) [inline src/sentry/search/snuba/backend.py:554]: Hopefully, this query (where message LIKE '%<query>%') on Groups table will be pared down enough to not scan a ton of data.

**Review by wedamija — COMMENTED** [review]
Any chance we could split up this refactor and the logic changes?

**Comment by barkbarkimashark** [comment]
> Any chance we could split up this refactor and the logic changes?

Dang! Thought I could get these two changes sneaked into on PR. Fair enough. Will do.

File: src/sentry/search/snuba/executors.py [hunk 2]
```diff
@@ -310,23 +365,6 @@ def snuba_search(
 
         return [(row["group_id"], row[sort_field]) for row in rows], total  # type: ignore[literal-required]
 
-    @staticmethod
-    def _transform_converted_filter(
```
Comment (barkbarkimashark) [inline src/sentry/search/snuba/executors.py:314]: Did a `find usage` on this and it was never used.

File: src/sentry/search/snuba/executors.py [hunk 3]
```diff
@@ -173,122 +288,62 @@ def snuba_search(
...
+            # remove any search_filters that are only available in postgres
+            # We special case date
+            if not (sf.key.name in self.postgres_only_fields or sf.key.name == "date")
+        ]
 
-        selected_columns = []
         if get_sample:
-            query_hash = md5(json.dumps(conditions).encode("utf-8")).hexdigest()[:8]
-            selected_columns.append(["cityHash64", [f"'{query_hash}'", "group_id"], "sample"])
-            sort_field = "sample"
```
Comment (wedamija) [inline src/sentry/search/snuba/executors.py:246]: `sort_field` is used later on, I think we need to keep this here

Comment (barkbarkimashark) [reply src/sentry/search/snuba/executors.py:246]: damn .. nice catch.

**Review by wedamija — APPROVED** [review]

File: src/sentry/issues/search.py [hunk 4]
```diff
@@ -42,52 +56,76 @@ class MergeableRow(TypedDict, total=False):
...
     environments: Optional[Sequence[Environment]],
     conditions: Sequence[Any],
 ) -> Optional[SnubaQueryParams]:
     organization = Organization.objects.filter(id=organization_id).first()
-    if (
-        organization
-        and features.has("organizations:performance-issues", organization)
-        and (not group_categories or GroupCategory.PERFORMANCE in group_categories)
-    ):
+    if organization and features.has("organizations:performance-issues", organization):
```
[the remark this answers is not in the document]
Comment (barkbarkimashark) [reply src/sentry/issues/search.py:128]: Yup, exactly. Since further up we check the flag.

**Pull request merged**